### PR TITLE
[SD-663] Add missing values for PDP Structured Data

### PR DESCRIPTION
**Pull request by pmilewski**

### CODE REVIEW

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 1]
```diff
@@ -9,10 +9,10 @@ def spree_breadcrumbs(taxon, separator = '', product = nil)
       return '' if current_page?('/') || taxon.nil?
 
       separator = raw(separator)
-      crumbs = [content_tag(:li, content_tag(:span, link_to(content_tag(:span, Spree.t(:home), itemprop: 'name'), spree.root_path, itemprop: 'url') + separator, itemprop: 'item'), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', class: 'breadcrumb-item')]
+      crumbs = [content_tag(:li, content_tag(:span, link_to(content_tag(:span, Spree.t(:home), itemprop: 'name'), spree.root_path, itemprop: 'url') + separator, itemprop: 'item'), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: spree.root_path,  class: 'breadcrumb-item', position: 0)]
       if taxon
         crumbs << taxon.ancestors.where.not(parent_id: nil).map { |ancestor| content_tag(:li, content_tag(:span, link_to(content_tag(:span, ancestor.name, itemprop: 'name'), seo_url(ancestor, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', class: 'breadcrumb-item') } unless taxon.ancestors.empty?
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb]: `itemid` and `position` needed here as well

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 2]
```diff
@@ -9,10 +9,10 @@ def spree_breadcrumbs(taxon, separator = '', product = nil)
...
-        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement')
+        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(taxon, params: permitted_product_params), position: 1)
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb]: @pmilewski this cannot be `position: 1`, the line above you've got an iteration of taxon ancestors, so this should be a  subsequent number based on the number of ancestors eg `position: taxon.ancestors.count + 1`

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 3]
```diff
@@ -9,10 +9,10 @@ def spree_breadcrumbs(taxon, separator = '', product = nil)
...
+        crumbs << taxon.ancestors.where.not(parent_id: nil).map.with_index { |ancestor, index| content_tag(:li, content_tag(:span, link_to(content_tag(:span, ancestor.name, itemprop: 'name'), seo_url(ancestor, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(ancestor, params: permitted_product_params), class: 'breadcrumb-item', position: index + 1) } unless taxon.ancestors.empty?
+        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(taxon, params: permitted_product_params), position: taxon.ancestors.where.not(parent_id: nil) + 1)
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb]:
```suggestion
        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(taxon, params: permitted_product_params), position: taxon.ancestors.where.not(parent_id: nil).count + 1)
```

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 4]
```diff
@@ -9,10 +9,10 @@ def spree_breadcrumbs(taxon, separator = '', product = nil)
...
-        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement')
+        crumbs << taxon.ancestors.where.not(parent_id: nil).map.with_index { |ancestor, index| content_tag(:li, content_tag(:span, link_to(content_tag(:span, ancestor.name, itemprop: 'name'), seo_url(ancestor, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(ancestor, params: permitted_product_params), class: 'breadcrumb-item', position: index + 1) } unless taxon.ancestors.empty?
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb]:
```suggestion
        crumbs << taxon.ancestors.where.not(parent_id: nil).map.with_index { |ancestor, index| content_tag(:li, content_tag(:span, link_to(content_tag(:span, ancestor.name, itemprop: 'name'), seo_url(ancestor, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(ancestor, params: permitted_product_params), class: 'breadcrumb-item', position: index + 1) } if taxon.ancestors.any?
```

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 5]
```diff
@@ -9,10 +9,10 @@ def spree_breadcrumbs(taxon, separator = '', product = nil)
...
+        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(taxon, params: permitted_product_params), position: taxon.ancestors.where.not(parent_id: nil) + 1)
         crumbs << content_tag(:li, content_tag(:span, content_tag(:span, product.name) + separator), class: 'breadcrumb-item') if product
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb]: @pmilewski shouldn't we add position here as well plus itemid?

**Comment by pmilewski** [comment]
https://github.com/spree/spree/pull/9815/files

**Pull request closed without merging**